src/rtctools_heat_network/esdl/esdl_additional_vars_mixin.py

```diff
@@ -0,0 +1,65 @@
+import esdl
+
+import numpy as np
+
+from rtctools.optimization.collocated_integrated_optimization_problem import (
+    CollocatedIntegratedOptimizationProblem,
+)
+
+
+class ESDLAdditionalVarsMixin(CollocatedIntegratedOptimizationProblem):
+
+    __temperature_options = {}
+
+    def pipe_classes(self, p):
+        return self._override_pipe_classes.get(p, [])
+
+    def esdl_heat_model_options(self):
+        """Overwrites the fraction of the minimum tank volume"""
+        options = super().esdl_heat_model_options()
+        options["min_fraction_tank_volume"] = 0.0
+        return options
+
+    def temperature_regimes(self, carrier):
+
+        temperature_options = []
+        temperature_step = 2.5
+
+        try:
+            temperature_options = self.__temperature_options[carrier]
+        except KeyError:
+            for asset in [
+                *self.heat_network_components.get("source", []),
+                *self.heat_network_components.get("ates", []),
+                *self.heat_network_components.get("buffer", []),
+                *self.heat_network_components.get("heat_pump", []),
+                *self.heat_network_components.get("heat_exchanger", []),
+                *self.heat_network_components.get("demand", []),
+            ]:
+                esdl_asset = self.esdl_assets[self.esdl_asset_name_to_id_map[asset]]
+                parameters = self.parameters(0)
+                for i in range(len(esdl_asset.attributes["constraint"].items)):
+                    constraint = esdl_asset.attributes["constraint"].items[i]
+                    if (
+                        constraint.name == "supply_temperature"
+                        and carrier == parameters[f"{asset}.T_supply_id"]
+                    ) or (
+                        constraint.name == "return_temperature"
+                        and carrier == parameters[f"{asset}.T_return_id"]
+                    ):
+                        lb = constraint.range.minValue
+                        ub = constraint.range.maxValue
```

Comment:
Assets could have conflicting temperature ranges, how to tackle, when looping over the assets?
Maybe something like:
 lb = constraint.range.minValue if constraint.range.minValue>lb else lb
To ensure that the lower bound does not become too low for some assets
ub = constraint.range.maxValue if constraint.range.maxValue<ub else ub
To ensure that the upper bound of the temperature does not become too high for some assets.

Reply:
agree, changed it, had to include a try statement as lb & ub might not yet exist